### ISSUE TITLE
Baudrate setting incorrectly applied

### ISSUE THREAD

**Issue by fgrfl** [vovkos/ioninja-issues#17]
The "Serial" Session only succeds in setting of baudrates of 115200 or lower, and only the ones in the dropdown list.
It allows setting all baudrates and shows them accordingly in the log, but the baudrate that is actually set is wrong.
This can be validated by opening a "Serial Monitor" Session on the same serial port.
Here it shows that the baudrate 38400 is set instead when setting a high or nonstandard baudrate in the "Serial" session.
Using a different program to open the serial port works with higher baudrates, but they are shown incorrectly in "Serial Monitor", e.g. "Set baud rate: 3263 bps"

Example:
* Set up an independent serial transmitter with guaranteed 230400bps sending human readable text periodically
* Connect it to the Rx line of e.g. /dev/ttyUSB0
* Open /dev/ttyUSB0 in "Serial Monitor" session
* Open /dev/ttyUSB0 in "Serial" session with baudrate 57600
* "Serial" session prints "Opened port /dev/ttyUSB0 baud: 57600..." and then garbage (expected with 230400bps data)
* "Serial Monitor" session prints "Port opened"..."Set baud rate: 57600 bps" and prints the garbage data as well
* Change baudrate to 230400 in "Serial" session
* "Serial" session prints "Set baud rate: 230400 bps" and then different looking garbage (not expected)
* "Serial Monitor" session prints "Set baud rate: 38400 bps" and the same garbage as "Serial" session
* Open serial port with a different program, e.g. gtkterm or cutecom, set baudrate to 230400
* "Serial Monitor" session prints "Port opened"..."Set baud rate: 3263 bps" and now outputs the correct data

I see 2 related problems: 
* "Serial" session not setting the correct baudrate when a baudrate higher then 115200 is applied (but 38400 instead)
* "Serial Monitor" session indicating a wrong baudrate when set with a different program, even though the baudrate set in that program is correctly applied. The indicated baudrate seems to be different for different programs setting that baudrate, but the same for different higher baudrates.

When selecting a nonstandard baudrate like 57601bps, both IONinja and other programs fail the same way (garbage receiving, baudrate 38400 indicated), so this could be a limitation of the usb/serial converter or serial driver. I did not validate the resulted physical baudrate.

TLDR:
I cannot apply 230400 bps in IONinja, but with a different serial program it works.

**Comment by vovkos** [comment]
Thank you for the bug report!

The issue is confirmed (it was reported via the support channel by other users, as well). Basically, all non-standard POSIX baud rates result in a fallback to 38400. The support for non-POSIX baud rates is implemented, but it has not been released yet. However, I can promise that it will be included in the next major release of IO Ninja tentatively scheduled for April-May.

Thanks again!

**Comment by vovkos** [comment]
FYI, the Windows version doesn't have this issue (so as a workaround you can run IO Ninja from a Windows VM)

**Comment by fgrfl** [comment]
Any updates on the schedule of the April-May release?

**Comment by vovkos** [comment]
> Any updates on the schedule of the April-May release?

Touche ;)

The release of this major update had to be postponed a few times for different reasons; apologies for that. But it's finally happening -- this week! I will post here as soon as it's out.

**Comment by fgrfl** [comment]
> The release of this major update had to be postponed a few times for different reasons

Well that happens sometimes. Thanks for the heads up, looking forward to it.

**Comment by vovkos** [comment]
The new IO Ninja is out! You can download it from the official website https://ioninja.com; please let me if it works for you.

**Comment by fgrfl** [comment]
The old version of IONinja does not show the new version under Help->About->Check for Updates on Windows as well as on Linux, also under linux with apt, it shows 3.15 as the latest version, however i installed https://tibbo.com/downloads/archive/ioninja/ioninja-5.0.1/ioninja-5.0.1-linux-amd64.deb manually. It was not immediatly obvious to me how to enter my existing licence, but I created a new account and could test it anyway: Baudrates higher than 115200 and inbetween the standard baudrates are working now.

**Comment by vovkos** [comment]
Right, we didn't publish package update manifests or apt/rpm repos yet, that's why there are no notifications and 3.15.1 is still reported as the latest version. We will collect some preliminary feedback and then publish update manifests and apt/rpm repos.

Glad to hear the baud rate issue fix worked for you!

**Comment by vovkos** [comment]
Sorry, I forgot to address this one:

> It was not immediatly obvious to me how to enter my existing licence, but I created a new account and could test it anyway

There are no more licenses. In IO Ninja 5, users can either (a) cherry-pick and enable particular capabilities or (b) get a subscription to enable everything at once. 

But of course, we didn't simply void the existing licenses.  As an IO Ninja 3 license holder, you are entitled to receive the total amount spent on your license(s) in the form of so-called "credits". To get these credits, simply log in using the same email as you used for purchasing the license.

IO Ninja credits can pay for any capabilities or subscriptions. The amount returned for a personal license covers 20 months of subscription; alternatively, you can select a bunch of plugins and features that you mostly use and permanently enable those by spending your credits on corresponding capabilities.

You can find more about the new account/capability/subscription/workgroup ecosystem on the introduction page: https://ioninja.com/capabilities-intro.html

Feel free to follow up with any questions regarding this topic. As for the baud rate problem, I'm closing the issue.
